Add a precondition to prevent the parsing of a command that has itself as its subcommand

NicFontana opened the pull request:
This avoids the infinite recursion that causes a crash and shows the user a meaningful error message.

Fixes: https://github.com/apple/swift-argument-parser/issues/192

<!--
    Thanks for contributing to the Swift Argument Parser!

    If this pull request adds new API, please add '?template=new.md'
    to the URL to switch to the appropriate template.

    Before you submit your request, please replace the paragraph
    below with the relevant details, and complete the steps in the
    checklist by placing an 'x' in each box:
    
    - [x] I've completed this task
    - [ ] This task isn't completed
-->

### Checklist
- [x] I've added at least one test that validates that my change is working, if appropriate
- [x] I've followed the code style of the rest of the project
- [x] I've read the [Contribution Guidelines](CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary

NicFontana commented:
Should I add an example that shows that the precondition catches the wrong ```CommandConfiguration``` and show the error message?

natecook1000 commented:
Re: testing — we don't have a great way of testing crashes, unfortunately. You could add a new parser error case (capturing the duplicated command?), and throw that instead of using a precondition failure.

NicFontana commented:
Good point @natecook1000, I've changed the PR to fix the issue as you suggested and added a new test to show it's working.
Let me know if you do think that this still needs changes :)

natecook1000 commented:
@swift-ci Please test

natecook1000 commented:
@swift-ci Please test
